### DISCUSSION TITLE
Send event to rust fontend.

### DISCUSSION THREAD

**Discussion by lll9p** [tauri-apps/tauri#10215] [Q&A] [answered]
Thank you very much for this project, Tauri. 

I am using leptos and tauriv2. 

How can I send events from the backend to the frontend?

I use:
`app.emit("the-event-name", ()).unwrap();`
to send events, and  use: 

```
window_event_listener_untyped("the-event-name", |event| {
logging::log!("got event");
});
```
in leptos, but the frontend does not receive any.

What should I do? Thank you for your help!

**Comment by lll9p** [comment, marked as answer]
The answer is so simple! Just do it like below, thanks tauri!

``` Rust
#[wasm_bindgen]
extern "C" {
    #[wasm_bindgen(js_namespace = ["window", "__TAURI__", "event"])]
    async fn listen(event: &str, handler: &js_sys::Function) -> JsValue;
}
```

``` Rust
spawn_local(async move {
    let closure = Closure::<dyn FnMut(_)>::new(move |s: JsValue| {
        logging::log!("recv event: {:?}", s);
    });
    listen("the-event-name", closure.as_ref().unchecked_ref()).await;
    closure.forget();
});
```

**Reply by lennjf** [reply]
when I ask OpenAI, it say that "closure.forget();" is danger, would lead to memory leak, should I worry about this ?
